libass/ass_parse.c

```diff
@@ -627,7 +637,12 @@ char *parse_tag(ASS_Renderer *render_priv, char *p, char *end, double pwr)
         if (t1 == -1 && t4 == -1) {
             t1 = 0;
             t4 = render_priv->state.event->Duration;
-            t3 = t4 - t3;
+            // If we were to subtract a large negative t3, this would
+            // overflow. However, the mathematical result would be
+            // beyond the end of the event anyway (corresponding to no
+            // fade-out), so we may as well just set it to the exact end
+            // of the event without any change in functionality.
+            t3 = t4 - FFMAX(t3, 0);
```

Comment:
Sure. Maybe a bit bloated (noise vs. importance). I actually wish we could leave "unimportant" overflows (that can't cause memory errors) just untouched. Not sure if that's a reasonable opinion.